src/control/zoom/ZoomControl.cpp

```diff
@@ -105,35 +113,64 @@ void ZoomControl::zoomScroll(ZoomDirection zoomIn, utl::Point<double> zoomCenter
         this->setZoomFitMode(false);
     }
 
-    if (this->zoomSequenceStart == -1 || scrollCursorPosition != zoomCenter) {
-        scrollCursorPosition = zoomCenter;
-        startZoomSequence(zoomCenter);
-    }
+    double multiplier = 1.0 + this->zoomStepScroll;
+    double newZoomStep = zoomIn ? multiplier : 1.0 / multiplier;
```

Comment:
Please do not use inline if statements. They are hard to read and therefore error-prone.